compendium/DeclarativeServices/test/gtest/TestComponentConfigurationImpl.cpp

```diff
@@ -936,6 +936,64 @@ namespace cppmicroservices
             framework.WaitForStop(std::chrono::milliseconds::zero());
         }
 
+        TEST(ComponentConfigurationTests, TestConcurrentStop)
+        {
+            auto framework = cppmicroservices::FrameworkFactory().NewFramework();
+            framework.Start();
+            ASSERT_TRUE(framework);
+
+            auto context = framework.GetBundleContext();
+            ASSERT_TRUE(context);
+
+            test::InstallAndStartDS(context);
+
+            std::vector<std::string> bundlesToInstall
+                = { "DSGraph01", "DSGraph02", "DSGraph03", "DSGraph04", "DSGraph05", "DSGraph06", "DSGraph07" };
+            std::vector<cppmicroservices::ServiceReferenceU> interfaces;
+            std::vector<cppmicroservices::Bundle> installedBundles;
+
+            for (auto const& bundleName : bundlesToInstall)
+            {
+                installedBundles.emplace_back(::test::InstallAndStartBundle(context, bundleName));
+            }
+
+            interfaces.push_back(context.GetServiceReference<test::DSGraph01>());
+            interfaces.push_back(context.GetServiceReference<test::DSGraph02>());
+            interfaces.push_back(context.GetServiceReference<test::DSGraph03>());
+            interfaces.push_back(context.GetServiceReference<test::DSGraph04>());
+            interfaces.push_back(context.GetServiceReference<test::DSGraph05>());
+            interfaces.push_back(context.GetServiceReference<test::DSGraph06>());
+            interfaces.push_back(context.GetServiceReference<test::DSGraph07>());
+
+            for (size_t i = 0; i < interfaces.size(); ++i)
+            {
+                auto sref = interfaces[i];
+                ASSERT_TRUE(static_cast<bool>(sref));
+                auto service = context.GetService(sref);
+                ASSERT_NE(service, nullptr);
+            }
+
+            EXPECT_NO_THROW({
+                std::thread bundleT = std::thread(
+                    [&installedBundles]()
+                    {
+                        for (auto& bundle : installedBundles)
+                        {
+                            bundle.Stop();
+                        }
+                    });
+                std::thread frameworkT = std::thread(
+                    [&framework]()
+                    {
+                        framework.Stop();
+                        framework.WaitForStop(std::chrono::milliseconds::zero());
+                    });
+
+                bundleT.join();
+                frameworkT.join();
```

Comment:
I looked at this. This function (as far as I see) is running the same function x number of times concurrently, whereas we just want to run two functions concurrently with eachother. I suppose we could run the `bundle.Stop()` functionality concurrently so that all bundles are stopping at the same time as eachother, but I am not sure that is necessary.